compiler/rustc_data_structures/src/sync.rs

```diff
@@ -26,6 +26,36 @@ use std::panic::{catch_unwind, resume_unwind, AssertUnwindSafe};
 pub use std::sync::atomic::Ordering;
 pub use std::sync::atomic::Ordering::SeqCst;
 
+mod mode {
+    use super::Ordering;
+
+    const UNINITIALIZED: usize = 0;
+    const SINGLE_THREAD: usize = 1;
+    const MULTIPLE_THREADS: usize = 2;
+
+    static PARALLEL: std::sync::atomic::AtomicUsize =
+        std::sync::atomic::AtomicUsize::new(UNINITIALIZED);
+
+    #[inline]
+    pub fn is_parallel() -> bool {
```

Comment:
Maybe `active` is a better name with it being re-exported in the `sync` module making it `sync::active()`?

Reply:
Yea I agree